net/connUDP.go

```diff
@@ -38,8 +132,63 @@ func (c *ConnUDP) Close() error {
 	return c.connection.Close()
 }
 
-// WriteContext writes data with context.
+func (c *ConnUDP) writeMulticastWithContext(ctx context.Context, udpCtx *ConnUDPContext, buffer []byte) error {
+	if udpCtx == nil {
+		return fmt.Errorf("cannot write multicast with context: invalid udpCtx")
+	}
+	if _, ok := c.packetConn.(*packetConnIPv4); ok && isIPv6(udpCtx.raddr.IP) {
+		return fmt.Errorf("cannot write multicast with context: invalid destination address")
+	}
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return fmt.Errorf("cannot write multicast with context: cannot get interfaces for multicast connection: %v", err)
+	}
+
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	for _, iface := range ifaces {
+		written := 0
+		for written < len(buffer) {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			default:
+			}
+
+			if err := c.packetConn.SetMulticastInterface(&iface); err != nil {
+				// log.Errorf("cannot set multicast interface %v for connection: %v", iface.Name, err)
```

Comment:
Remove the stale comment, or pass errors to a callback `func(error)`? Same below.

Reply:
removed